Cargo.toml

```diff
@@ -9,8 +9,5 @@ repository = "https://github.com/async-rs/stop-token"
 description = "Experimental cooperative cancellation for async-std"
 
 [dependencies]
-pin-project-lite = "0.1.0"
-async-std = "1.0"
-
-[features]
-unstable = ["async-std/unstable"]
+pin-project-lite = "0.2.0"
+async-std = "1.8"
```

Comment:
I'm curious about this change.  If I understand [cargo dependency resolution](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html), `async-std = "1.0"` is short for `async-std="^1.0"`, which in this case would resolve to `async-std="1.9"` since: `^1      :=  >=1.0.0, <2.0.0`.  So I think this version bump is superfluous so semver will evaluate to the latest `1.x` release, namely `1.9`.

Reply:
1.8 is the earliest version this can support.  setting it like this is equivalent to `^1.8` or `>=1.8.0 <2.0.0`.  As this is a library that means users can themselves say for example `async-std = "^1.18"` in their own cargo.toml and this library will not force another duplicate version of async-std to be compiled in as that version is still compatible here.
If we would set this to `^1` then someone depending on this could say `= 1.3.0` and we'd get that version here and it would break.

Reply:
I think i'm missing something about dependency resolution here.  When I set `async-std="1.0"` cargo downloads `1.9`.  I guess I don't understand how to tell cargo to use less than `1.9`.

Reply:
your binary says `async-std >= 1.0.0, < 2.0.0` and you also depend on stop-token which says `async-std >= 1.8.0, < 2.0.0`.  As that's the only input to cargo's dependency resolver it decides it also prefers the most recent version and it resolves this to `async-std 1.9.0` which satisfies all those constraints.

If you want cargo to use something older than 1.9 you need to tell it to do that in your Cargo.toml.  Set `async-std: <1.9` probably with some other constraints added. If you are asking for `async-std < 1.8` but do ask for `stop-token >= 0.2` then cargo will build 2 versions of async-std into your binary.